terra-jupyter-python/Dockerfile

```diff
@@ -20,69 +20,82 @@ RUN apt-get update && apt-get install -yq --no-install-recommends \
 
 ENV HTSLIB_CONFIGURE_OPTIONS="--enable-gcs"
 
+# Dev note: in general, do not pin Python packages to any particular version.
+# Depend on the smoke tests to help us identify any package incompatibilties.
+#
+# If we find that we do need to pin a package version, be sure to:
+# 1) Add a comment saying what needs to be true for us to remove the pin.
+#    (e.g. link to an issue and put the details there)
+# 2) If the smoke tests did not show the problem, add a new test case to improve
+#    test coverage for the identified problem.
 RUN pip3 -V \
  && pip3 install --upgrade pip \
- && pip3 install numpy==1.15.2 \
- && pip3 install py4j==0.10.7 \
- && python3 -mpip install matplotlib==3.0.0 \
- && pip3 install pandas==0.25.3 \
- && pip3 install pandas-gbq==0.12.0 \
- && pip3 install pandas-profiling==2.4.0 \
- && pip3 install seaborn==0.9.0 \
- && pip3 install python-lzo==1.12 \
- && pip3 install google-cloud-bigquery==1.23.1 \
- && pip3 install google-api-core==1.6.0 \
- && pip3 install google-cloud-bigquery-datatransfer==0.4.1 \
- && pip3 install google-cloud-datastore==1.10.0 \
- && pip3 install google-cloud-resource-manager==0.30.0 \
- && pip3 install google-cloud-storage==1.23.0 \
- && pip3 install scikit-learn==0.20.0 \
- && pip3 install statsmodels==0.9.0 \
- && pip3 install ggplot==0.11.5 \
- && sed -i 's/pandas.lib/pandas/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
- # the next few `sed` lines are workaround for a ggplot bug. See https://github.com/yhat/ggpy/issues/662
- && sed -i 's/pandas.tslib.Timestamp/pandas.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
- && sed -i 's/pd.tslib.Timestamp/pd.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/stats/smoothers.py \
- && sed -i 's/pd.tslib.Timestamp/pd.Timestamp/g' /usr/local/lib/python3.7/dist-packages/ggplot/utils.py \
- && pip3 install bokeh==1.0.0 \
- && pip3 install pyfasta==0.5.2 \
- && pip3 install markdown==2.4.1 \
- && pip3 install pdoc3==0.7.2 \
- && pip3 install biopython==1.72 \
- && pip3 install bx-python==0.8.2 \
- && pip3 install fastinterval==0.1.1 \
- && pip3 install matplotlib-venn==0.11.5 \
- && pip3 install bleach==1.5.0 \
- && pip3 install cycler==0.10.0 \
- && pip3 install h5py==2.7.1 \
- && pip3 install html5lib==0.9999999 \
- && pip3 install joblib==0.11 \
- && pip3 install keras==2.1.6 \
- && pip3 install patsy==0.4.1 \
- && pip3 install protobuf==3.7.1 \
- && pip3 install pymc3==3.10.0 \
- && pip3 install pyparsing==2.2.0 \
+ && pip3 install numpy \
```

Comment:
Per offline discussion I'd probably use a more explicit version lockfile of some kind (which can be easily regenerated) for the sake of determinism, but just my 2c - this doesn't directly affect me or the AoU image.

Reply:
Thank you @calbach .  Leonardo team, what is your preference?

My goal with this PR is for the resulting images to have newer versions of all Python packages. Happy to change the details for how that is accomplished.

Reply:
I'm willing to try unpinning the versions

Reply:
Unpinning the versions sounds good to me too, we can do additional testing in Leonardo before releasing